src/framework/core/js/DataBinding.js

```diff
@@ -409,91 +421,150 @@ https://github.com/fluid-project/infusion/raw/main/Infusion-LICENSE.txt
         return (reca.completeOnInit ? 1 : 0) - (recb.completeOnInit ? 1 : 0);
     };
 
+    fluid.subscribeResourceModelUpdates = function (that, resourceMapEntry) {
+        var treeTransaction = fluid.currentTreeTransaction();
+        var resourceSpec = resourceMapEntry.resourceSpec;
+        var resourceUpdateListener = function () {
+            // We can't go for currentTreeTransaction() in this listener because we in the "dead space" between workflow
+            // functions where it has not been restored by the waitIO listener. Isn't the stack a sod.
+            var initTransaction = fluid.getImmediate(treeTransaction, ["initModelTransaction", that.id]);
+            var trans = initTransaction ? initTransaction.transaction : that.applier.initiate();
+            resourceMapEntry.listeners.forEach(function (oneListener) {
+                var innerValue = fluid.getImmediate(resourceSpec, oneListener.resourceSegs);
+                var segs = oneListener.segs;
+                trans.change(segs, null, "DELETE");
+                trans.change(segs, innerValue);
+            });
+            if (!initTransaction) {
+                trans.commit();
+            } else {
+                var transRec = fluid.getModelTransactionRec(fluid.rootComponent, trans.id);
+                fluid.clearLinkCounts(transRec, true);
+                that.applier.preCommit.fire(trans, that);
+            }
+        };
+        resourceSpec.onFetched.addListener(resourceUpdateListener);
+        fluid.recordListener(resourceSpec.onFetched, resourceUpdateListener, fluid.shadowForComponent(that));
+    };
+
+    fluid.resolveResourceModelWorkflow = function (shadows, treeTransaction) {
+        var initModelTransaction = treeTransaction.initModelTransaction;
+        // TODO: Original comment from when this action was in operateInitialTransaction, now incomprehensible
+        // Do this afterwards so that model listeners can be fired by concludeComponentInit
```

Comment:
I'm not really sure what this TODO means.

Reply:
Yes, I'm not sure what it means either - it has been left here in case something jogs memory later on, or there is the time to puzzle it out